sdk/core/src/policies/retry_policies/retry_policy.rs

```diff
@@ -61,29 +61,33 @@ where
                 }
                 Ok(response) => {
                     // Error status code
-                    let error = HttpError::new(response).await;
-                    let status = StatusCode::from_u16(error.status()).unwrap();
+                    let code = response.status().as_u16();
+
+                    let http_error = HttpError::new(response).await;
+                    let status = StatusCode::from_u16(code);
                     let error = Error::full(
                         ErrorKind::http_response(
-                            status.as_u16(),
-                            error.error_code().map(|s| s.to_owned()),
+                            code,
+                            http_error.error_code().map(|s| s.to_owned()),
                         ),
-                        error,
+                        http_error,
                         "server returned error status which will not be retried",
                     );
 
-                    if !RETRY_STATUSES.contains(&status) {
-                        log::error!(
-                            "server returned error status which will not be retried: {}",
+                    if let Ok(status) = status {
```

Comment:
Hmm I'm not sure this is better. If the status getting returned is not able to be turned into a `Status` type, that's a bug on our part. I think panicking here is actually the right thing to do.

Reply:
The response parsing context here is handling a pre-defined set of status codes that should require retrying the request.

My thought is that there are valid HTTP status codes that `http::StatusCode` does not support, including 425 Early Data.   Instead of panicking in that case, this instead does not retry.

Of note, we're not actually returning the StatusCode, just using it as a method to let users set how we respond to specific status codes.

https://www.rfc-editor.org/rfc/rfc8470.html

Reply:
> My thought is that there are valid HTTP status codes that http::StatusCode does not support, including 425 Early Data. Instead of panicking in that case, this instead does not retry.

Hmm... that's a real shame and seems like a bug in the `http` crate. I wonder if it would be better to just get rid of the use of the `StatusCode` type completely...

Reply:
Revisiting this today, I think this is a non-issue.

The underlying libraries used to parse the response use `http::StatusCode`, which means the value has to be valid in this context anyways.

Also relevant, this isn't a user-facing feature.  The SDK does not expose specifying a custom set of status codes to retry.  In this context, we only use `http:;StatusCode` as a convenience mechanism.  It's easier to read remember `StatusCode::REQUEST_TIMEOUT` than the number 408 and a comment.

I'll update this code to use expect, such that it's clear we this should not ever fail.